### PR TITLE
Fix bugs in FY5253Quarter and LastWeekOfMonth

### PR DESCRIPTION
**Pull request by jbrockmendel**
Similar to [need to lookup reference], this fixes bugs in `LastWeekOfMonth` and `FY5253Quarter` to ensure that the fast implementation `offset.onOffset(ts)` matches the slow (base class) implementation `(ts + offset) - offset == ts`.

I haven't yet looked at BusinessHour and CustomBusinessHour, but other than that I think these are the last classes that have this category of bug.

- [ ] closes #xxxx
- [x] tests added / passed
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] whatsnew entry

### CODE REVIEW

File: pandas/tests/tseries/offsets/test_fiscal.py [hunk 1]
```diff
@@ -633,3 +633,23 @@ def test_fy5253_nearest_onoffset():
     fast = offset.onOffset(ts)
     slow = (ts + offset) - offset == ts
     assert fast == slow
+
+
+def test_fy5253qtr_onoffset_nearest():
+    ts = Timestamp('1985-09-02 23:57:46.232550356-0300',
```
Comment (gfyoung) [inline pandas/tests/tseries/offsets/test_fiscal.py]: Reference your PR as a comment.  Same for all other tests that you added.

**Review by gfyoung — APPROVED** [review]

**Comment by gfyoung** [comment]
https://travis-ci.org/pandas-dev/pandas/jobs/324352177

Not sure why this keeps timing out.

File: pandas/tseries/offsets.py [hunk 2]
```diff
@@ -1982,46 +1972,76 @@ def _offset(self):
...
+        else:
+            tdelta = Timedelta(0)
+
+        # Note: we always have tdelta.value >= 0
+        return start, num_qtrs, tdelta
+
     @apply_wraps
     def apply(self, other):
-        base = other
+        # Note: self.n == 0 is not allowed.
```
Comment (jreback) [inline pandas/tseries/offsets.py]: you allow this below

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/tseries/offsets.py (hunk 2 above)
Comment (jbrockmendel) [reply pandas/tseries/offsets.py]: This is enforced in init.

File: pandas/tseries/offsets.py [hunk 3]
```diff
@@ -1951,7 +1941,7 @@ class FY5253Quarter(DateOffset):
         or 14 week when needed. {1,2,3,4}
     variation : str
         {"nearest", "last"} for "LastOfMonth" or "NearestEndMonth"
-    """
+    """ % _5253doc
```
Comment (jreback) [inline pandas/tseries/offsets.py]: can you use ``.format()``

Comment (jreback) [reply pandas/tseries/offsets.py]: and same on all the templates (generally we want to always use the .format())

File: pandas/tseries/offsets.py [hunk 4]
```diff
@@ -1982,46 +1972,76 @@ def _offset(self):
...
+        elapsed between that year-end and other, and the remaining Timedelta
+        since the most recent quarter-end.
+
+        Parameters
+        ----------
+        other : datetime or Timestamp
+
+        Returns
+        -------
+        prev_year_end : Timestamp giving most recent fiscal year end
```
Comment (jreback) [inline pandas/tseries/offsets.py]: tuple of

**Review by jreback — CHANGES_REQUESTED** [review]
small comments, needs a rebase, ping on green.

File: pandas/tseries/offsets.py (hunk 3 above)
Comment (jbrockmendel) [reply pandas/tseries/offsets.py]: That was the first attempt.  That syntax doesn't play nicely with the set syntax inside the docstring.  The docstring change here is tangential to the PR; I'll just revert it.

**Comment by jbrockmendel** [comment]
Ping

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks!

**Pull request merged**